neoload/commands/report.py

```diff
@@ -0,0 +1,1056 @@
+    if summary_precludes_details_fetch(data): return
+    if components['transactions']:
+        gprint("Getting transactions...")
+
+        json_elements_transactions = rest_crud_get(get_end_point(__id, __operation_elements) + "?category=TRANSACTION")
+        if not elements_filter is None:
+            json_elements_transactions = filter_elements(json_elements_transactions, elements_filter)
+
+        json_elements_transactions = get_elements_data(__id, json_elements_transactions, time_binding, True, statistics_list, use_txn_raw)
+        json_elements_transactions = list(sorted(json_elements_transactions, key=lambda x: x['display_name']))
+
+        data['elements']['transactions'] = json_elements_transactions
+
+def fill_single_monitors(__id, components, data):
+    if summary_precludes_details_fetch(data): return
+    if components['ext_data'] or components['controller_points']:
+        gprint("Getting monitors...")
+        data['monitors'] = rest_crud_get(get_end_point(__id, __operation_monitors))
+
+def fill_single_ext_data(__id, components, data):
+    if summary_precludes_details_fetch(data): return
+    if components['ext_data']:
+        ext_datas = get_mon_datas(__id, lambda m: m['path'][0] == 'Ext. Data', data['monitors'], True)
+        ext_datas = list(sorted(ext_datas, key=lambda x: x['display_name']))
+        data['ext_data'] = ext_datas
+
+def fill_single_controller_points(__id, components, data):
+    if summary_precludes_details_fetch(data): return
+    if components['controller_points']:
+        data['controller_points'] = get_mon_datas(__id, lambda m: m['path'][0] == 'Controller', data['monitors'], True)
+
+# examples
+# --filter "timespan=2m-4m"
+# --filter "timespan=10%-90%"
+# --filter "timespan=5m15s;results=cur|-2;elements=11a783ad-ae93-4f05-8517-3fd2feb8452d|S05_Checkout|(.*)Login"
+def fill_time_binding(time_binding):
+    try:
+        timespan_spec = time_binding["time_filter"] if "time_filter" in time_binding else None
+        if not timespan_spec is None:
+            total_duration_sec = time_binding["summary"]["duration"] / 1000
+            time_parts = timespan_spec.split("-")
+            from_spec = time_parts[0] if len(time_parts[0].strip())>0 else None
+            to_spec = time_parts[1] if len(time_parts) > 1 and len(time_parts[1].strip())>0 else None
+            from_secs = 0 if from_spec is None else translate_time_part_to_seconds(total_duration_sec, from_spec)
+            to_secs = total_duration_sec if to_spec is None else translate_time_part_to_seconds(total_duration_sec, to_spec)
+            time_binding['from_secs'] = round(from_secs)
+            time_binding['to_secs'] = round(to_secs)
+        return time_binding
+    except Exception:
+        raise ValueError("Something went wrong while fill_time_binding")
+
+
+time_part_mod_to_sec = {
+    "h": lambda x: x * 60 * 60,
+    "m": lambda x: x * 60,
+    "s": lambda x: x
+}
```

Comment:
It's forbidden to declare global variables in the middle of the file. Because we don't see it, and we don't understand why they need to be global.
Make it a private local constant in the function _translate_time_part_to_seconds_

Reply:
266dbca moved to local variable instead of global; don't want to create a separate constants module just for this